core/base/mergeTreePrincipalGeodesics/MergeTreePrincipalGeodesics.cpp

```diff
@@ -0,0 +1,45 @@
+#include <MergeTreePrincipalGeodesics.h>
+
+namespace ttk {
+  double MergeTreePrincipalGeodesics::verifyOrthogonality(
+    std::vector<std::vector<std::vector<double>>> &vS,
+    std::vector<std::vector<std::vector<double>>> &v2s,
+    bool doPrint) {
+    unsigned int geodesicNumber = vS.size() - 1;
+    if(geodesicNumber == 0)
+      return 0.0;
+    if(doPrint)
+      printMsg("Scalar Products:");
+    double cost = 0.0;
+    std::vector<std::vector<double>> sumVs;
+    ttk::Geometry::multiAddVectorsFlatten(vS, v2s, sumVs);
+    for(unsigned int i = 0; i < geodesicNumber; ++i) {
+      double scalarProd
+        = ttk::Geometry::dotProduct(sumVs[i], sumVs[geodesicNumber]);
+      cost += std::pow(scalarProd, 2);
```

Comment:
please avoid using `std::pow` when the exponent is an integer.
instead, thanks for using @pierre-guillou faster lambdas (`TTK_POW_LAMBDA`). this comes with performance improvements. thanks for double-checking your code for other occurences.

Reply:
Modified in the next commit!
As suggested by @pierre-guillou I simply used a multiplication, since the power is fixed, the call to `TTK_POW_LAMBDA` could be less efficient than just a multiplication. The multiplication is still better than `std::pow`.